src/org/rascalmpl/uri/URIUtil.java

```diff
@@ -433,6 +433,32 @@ public static ISourceLocation removeOffset(ISourceLocation prev) {
     public static ISourceLocation createFromURI(String value) throws URISyntaxException {
         return vf.sourceLocation(createFromEncoded(value));
     }
+
+	/**
+	 * Extracts the extension (the characters after the last . in the file name),
+	 * if any, and otherwise returns the empty string.
+	 * @param loc
+	 * @return
+	 */
+	public static String getExtension(ISourceLocation loc) {
+		String path = loc.getPath();
+		boolean endsWithSlash = path.endsWith(URIUtil.URI_PATH_SEPARATOR);
+		if (endsWithSlash) {
```

Comment:
so if I ask the extension of the directory `/x/path/y.nl/` we would get `.nl` as extension?

Reply:
yes, as it should be. the final slash is a indicator that the file is a folder, but it is not a part of its name.